src/blockMonitor/blockProcessor.ts

```diff
@@ -57,6 +69,10 @@ export class BlockProcessor extends StartStopService {
     protected async startInternal(): Promise<void> {
         // Make sure the current head block is processed
         const initialBlockNumber = await this.provider.getBlockNumber();
+
+        // This should never happen
+        if (initialBlockNumber == null) throw new ApplicationError(`getBlockNumber returned ${initialBlockNumber}`);
```

Comment:
I left it because it happened in some tests, shouldn't really happen in the real world; anyway, since the provider is an external class, I would leave the check - at least it will give better feedback if that fails for any reason (e.g. further tests not properly mocking the provider!).

Reply:
If that's why you're leaving it you should leave a comment explaining why it happens in some tests. I would say this is a problem with the way we're doing testing though, so whether you leave the error here or not you should create a ticket that addresses the issue. Generally it would be better if we fix the problem with the tests rather than adding redundant code to the production code.

Reply:
Ok, then, I removed the extra check. It would be nice if TypeScript allowed strict runtime type checking as a compiler option; would be super handy for testing.